akka-actor/src/main/scala/akka/io/SimpleDnsCache.scala

```diff
@@ -15,7 +15,7 @@ class SimpleDnsCache extends Dns with PeriodicCacheCleanup {
 
   private val cache = new AtomicReference(new Cache(
     immutable.SortedSet()(ExpiryEntryOrdering),
-    immutable.Map(), clock))
+    immutable.Map(), clock _))
```

Comment:
So automatic use of a empty paren method as a function argument has been deprecated in 2.12 and you get this warning `Eta-expansion of zero-argument method values is deprecated`. So you need to add `_` to get `clock` the method unambigously.